File: cgi-bin/listdir.py
#!/usr/bin/python

import cgi
import os
import json

print "Content-Type: text/html; charset=UTF-8\n"

form = cgi.FieldStorage()
if "dir" not in form:
    print '{error:"Please specify the dir parameter"}'
    exit

root = os.getcwd()
dir = root + "/" + form.getlist("dir")[0]
contents = os.listdir(dir)
print json.dumps(contents)

File: cgi-bin/listdir.php
<?php
header('Content-Type: application/json');

if (!isset($_GET['dir'])) {
    echo '{"error":"Please specify the dir parameter"}';
    exit;
}

$dir = dirname(__DIR__) . DIRECTORY_SEPARATOR . $_GET['dir'];
$contents = glob($dir . DIRECTORY_SEPARATOR . '*');
$files = array();
foreach ($contents as $file) {
    $files[] = basename($file);
}
echo json_encode($files);
